src/testing/cluster.zig

```diff
@@ -300,28 +305,28 @@ pub fn ClusterType(comptime StateMachineType: fn (comptime Storage: type, compti
             }
         }
 
-        pub fn restart_replica(cluster: *Self, replica_index: u8) void {
+        /// Returns whether the replica was crashed.
+        /// Returns an error when the replica was unable to recover (open).
+        pub fn restart_replica(cluster: *Self, replica_index: u8) !void {
             assert(cluster.replica_health[replica_index] == .down);
 
+            // Pass the old replica's Time through to the new replica. It will continue to tick while
+            // the replica is crashed, to ensure the clocks don't desyncronize too far to recover.
+            try cluster.open_replica(replica_index, cluster.replicas[replica_index].time);
```

Comment:
My mental borrow-checker sounds alarm here: I think sometimes Zig sneakily converts by-value parameters into by-immutable-reference, and this might be problematic for the `Time` parameter here, as we are overwriting the whole replica in `open_replica`. 

Is there some sure-fire way to force a copy here?